src/overlays/actors/ovl_En_Minideath/z_en_minideath.c

```diff
@@ -41,147 +51,861 @@ ActorInit En_Minideath_InitVars = {
+        this->collider.base.atFlags |= AT_ON;
+        this->crowdState = true;
+
+        EnMinideath_CrowdParent(this, play);
+
+        for (i = 0; i < MINIDEATH_NUM_EFFECTS; i++) {
+            Math_Vec3f_Sum(&thisx->world.pos, &this->effects[i].vel, &this->effects[i].pos);
+        }
+
+        EnMinideath_SetupCrowdParent(this);
+    } else {
+        EnMinideath_SetupPreBattle(this);
+    }
+}
+
+void EnMinideath_Destroy(Actor* thisx, PlayState* play) {
+    EnMinideath* this = THIS;
+
+    Collider_DestroyJntSph(play, &this->collider);
+}
+
+void EnMinideath_DropCollectible(Vec3f* spawnPos, PlayState* play) {
+    if (sItemDropTimer == 0 && Rand_ZeroOne() < 0.3f) {
+        if (Rand_ZeroOne() < 0.5f) {
+            Item_DropCollectible(play, spawnPos, ITEM00_ARROWS_10);
+        } else {
+            Item_DropCollectible(play, spawnPos, ITEM00_MAGIC_JAR_BIG);
+        }
+        sItemDropTimer = 800; // wait 40 seconds before next drop
+    }
+}
+
+void EnMinideath_UpdateCrowdState(EnMinideath* this, s32 crowdParent) {
+    // If the state is changed to the same as before, don't update the count
+    if (this->crowdState != crowdParent) {
+        this->crowdState = crowdParent;
+
+        // Gomess params stores the number of crowding bats
+        if (crowdParent == true) {
+            this->actor.parent->params++;
+        } else {
+            this->actor.parent->params--;
+        }
+    }
+}
+
+void EnMinideath_RandomizeEffects(EnMinideath* this) {
+    MiniDeathEffect* effect;
+    s32 i;
+
+    for (effect = this->effects, i = 0; i < MINIDEATH_NUM_EFFECTS; i++, effect++) {
+        effect->vel.x = Rand_CenteredFloat(80.0f);
+        effect->vel.y = Rand_ZeroFloat(40.0f);
+        effect->vel.z = Rand_CenteredFloat(80.0f);
+        effect->animFrame = (s32)Rand_ZeroFloat(9.0f) % MINIDEATH_ANIM_LENGTH;
+        effect->angle.x = (s32)Rand_Next() >> 17;
+        effect->angle.y = (s32)Rand_Next() >> 16;
+        effect->state = 0;
+        effect->timer = 0;
+    }
+}
+
+void EnMinideath_UpdateEffects(EnMinideath* this, PlayState* play) {
+    s16 angle;
+    MiniDeathEffect* effect;
+    f32 phi_fv0;
+    f32 phi_fa0;
+    s32 i;
+    s32 phi_s7;
+    s32 phi_s3;
+
+    phi_s3 = 0;
+    phi_s7 = 0;
+
+    effect = this->effects;
+    for (i = 0; i < MINIDEATH_NUM_EFFECTS; i++, effect++) {
+        if (effect->state == 0) {
+            effect->vel.x += 3.0f * Math_CosS(effect->angle.x) * Math_SinS(effect->angle.y);
+            effect->vel.y += 3.0f * Math_SinS(effect->angle.x);
+            effect->vel.z += 3.0f * Math_CosS(effect->angle.x) * Math_CosS(effect->angle.y);
+
+            if (this->actionFunc == EnMinideath_PreBattle) {
+                phi_fv0 = 100.0f;
+                phi_fa0 = SQ(phi_fv0);
+            } else if (this->actionFunc == EnMinideath_StartSwarm || this->actionFunc == EnMinideath_SwarmLeader) {
+                phi_fv0 = 20.0f;
+                phi_fa0 = SQ(20.0f);
```

Comment:
```suggestion
                phi_fa0 = SQ(phi_fv0);
```
Does this work like above? Same for the `40.0f`. Or possibly change the one above to `SQ(100.0f)`?

Reply:
Unfortunately neither option to make these consistent works